rmf_task/src/rmf_task/Priority.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef SRC__RMF_TASK__PRIORITY_HPP
+#define SRC__RMF_TASK__PRIORITY_HPP
+
+#include <memory>
+
+#include <rmf_utils/impl_ptr.hpp>
+
+namespace rmf_task {
+
+/// Implement this for a new prioritization scheme.
+class Priority
+{
+public:
+
+  /// Get the value of this priority
+  virtual std::size_t value() const = 0;
+
+  virtual ~Priority() = default;
```

Comment:
After removing the virtual function above, we can change this to

```
virtual ~Priority() = 0
```

which is a C++ syntax trick to make this class a pure abstract class without having any pure virtual functions. Being a pure abstract class means it's impossible to create a `Priority` object directly; you can only create an object of a class derived from `Priority`.

Reply:
Thanks for explaining this. The compiler required me to add a function body for the pure virtual destructor so I've added that in `Priority.cpp` (that is the only contents of this file). 

https://github.com/osrf/rmf_core/pull/278/commits/68ec61f4cdee5eae420a912d29b750b00acb1826